src/controllers/plugins/PluginPermission.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#ifdef CHATTERINO_HAVE_PLUGINS
+
+#    include <vector>
+
+namespace chatterino {
+
+struct PluginPermission {
+    enum class Type {
+        FilesystemRead,
+        FilesystemWrite,
+    };
+    Type type;
+
+    std::vector<QString> errors;
+
+    bool isValid() const
+    {
+        return this->errors.empty();
+    }
+    QString toHtmlEscaped() const;
```

Comment:
Could this be `toHtml` instead? Some permissions might want to render information in HTML (like `"do something in <b>[html-escaped]</b>"`). There should probably be a newline above, too.